### DISCUSSION TITLE
Couple of questions...

### DISCUSSION THREAD

**Discussion by nradulaski-godaddy** [jscutlery/semver#536] [Q&A] [answered]
1. Is it possible to have commit message body also in changelogs?
2. When running `nx affected --target=version --parallel=1` is it possible to have just one commit for all lib bumps and than to run `postTargets` sequentially?
3. I used generator and `nx affected ...` (independent) works but `nx run workspace:version` throws ` >  NX   Cannot find project 'workspace'`. Where might I be mistaking?
4. With independent mode I expect number of tags to quickly skyrocket. Do you have any advice how to deal with it?

**Comment by edbzn** [comment, marked as answer]
Hi @nradulaski-godaddy,

> 1. Is it possible to have commit message body also in changelogs?

 Well, not actually as we use [conventional-changelog-angular](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-angular) (or [conventional-changelog-conventionalcommits](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-conventionalcommits) depending on the preset) to generate the changelog and it ignores the commit message body.

> 2. When running nx affected --target=version --parallel=1 is it possible to have just one commit for all lib bumps and than to run postTargets sequentially?

No, not actually, this is something we would like to tackle, but we need Nx to support global targets first. Here is the related issue https://github.com/jscutlery/semver/issues/61.

> 3. I used generator and nx affected ... (independent) works but nx run workspace:version throws > NX Cannot find project 'workspace'. Where might I be mistaking?

When installing semver using the `@jscutlery/semver:install` generator you have to choose between the sync or independent mode. If you choose independent mode then you can't run the sync mode as it's not configured by the installation generator.

> 4. With independent mode I expect number of tags to quickly skyrocket. Do you have any advice how to deal with it?

Yeah, true, it depends on your context, how many projects do you have? If these projects are totally decoupled and need to be versioned on their own then go for the independent mode, otherwise, the sync mode is what you are looking for. For instance, the [Babel repository](https://github.com/babel/babel/tree/main/packages) has many projects, and they are more or less coupled together, so here the sync mode makes sense (and that's what they used) because in the end it's one single product.

Hope it helps. Cheers.

**Reply by radulle** [reply]
Thank you for the answers. 
3. It's strange that I did not get asked to chose the sync/indi mode.

I can always use `--syncVersions` which is like sync mode, can't I?

4. Would it be bad just to delete old tags?

**Reply by edbzn** [reply]
Hi @radulle,

> It's strange that I did not get asked to chose the sync/indi mode

If you ran `nx generate @jscutlery/semver:install --syncVersions` then it doesn't ask you if you want to use the sync mode, but if you don't specify the option it will prompt. 

>  I can always use --syncVersions which is like sync mode, can't I?

Once you have installed the plugin with sync mode activeted, then simply run:
```
nx run workspace:version
```
no need to provide the `--syncVersions` option everytime, it's kept in your workspace definition.

> Would it be bad just to delete old tags?

It depends, why would you want to delete old tags? They were generated by semver?

**Reply by nradulaski-godaddy** [reply]
Yes, I'll have a large number of libs, so there will be lots of tags that get generated by semver (because each lib has it's own set of tags). Having ability to publish independently is nice.

**Reply by edbzn** [reply]
@nradulaski-godaddy Yeah that's why we want to have more control between "independent" and "sync", our next goal would be to handle versioning groups so you can "reduce" tags/commits overhead, but it is not for tomorrow. Let me know if I can help you more.
